runtime/curl.cpp

```diff
@@ -663,13 +667,25 @@ mixed f$curl_exec(curl_easy easy_id) noexcept {
     return false;
   }
 
+  if (kphp_tracing::is_turned_on()) {
+    string url = easy_context->get_info(CURLINFO_EFFECTIVE_URL).as_string();
+    kphp_tracing::on_curl_exec_start(easy_context->uniq_id, url);
+  }
+
   easy_context->cleanup_for_next_request();
   easy_context->error_num = dl::critical_section_call(curl_easy_perform, easy_context->easy_handle);
 
   if (easy_context->error_num != CURLE_OK && easy_context->error_num != CURLE_PARTIAL_FILE) {
+    if (kphp_tracing::is_turned_on()) {
+      kphp_tracing::on_curl_exec_fail(easy_context->uniq_id, -easy_context->error_num); // error_num > 0, pass negative
+    }
     return false;
   }
 
+  easy_context->bytes_recv = easy_context->received_data.size();
```

Comment:
I didn't know about this functionality. If curl can provide this info natively, it would be better, of course. Let's discuss this moment outside of Github discusttions